src/compiler/mod.rs

```diff
@@ -972,12 +972,57 @@ impl Compiler {
         }
     }
 
+    fn get_llvm_type_with_function_args(&self, typ: parser::Type, param_types: &[llvm::Type], return_type: llvm::Type) -> llvm::Type {
+        match typ {
+            parser::Type::Vector => self.context.double_type().pointer_type(0),
+            parser::Type::Numeric => self.context.double_type(),
+            parser::Type::Function => self
+                .context
+                .function_type(return_type, param_types, false)
```

Comment:
couldnt we call `get_lambda_func_args` and `get_lambda_func_return_type` here instead of passing it to `get_llvm_type_with_function_args` from outside(passing `arg` instead)?